sdk/instrumentation/opentelemetry-instrumentation-azure-sdk/src/transformations.ts

```diff
@@ -93,3 +93,18 @@ export function toSpanOptions(spanOptions?: InstrumenterSpanOptions): SpanOption
     links,
   };
 }
+
+/**
+ * Given an environment variable name, returns whether it is truthy.
+ *
+ * We consider "false" and "0" as falsey when used in environment variables.
+ *
+ * @internal
+ */
+export function environmentVariableToBoolean(environmentVariableName: string): boolean {
+  if (typeof process === "undefined") {
+    return false;
+  }
+  const valueWithDefault = (process.env[environmentVariableName] ?? "").toLowerCase();
```

Comment:
if I'm reading [this issue](https://github.com/nodejs/node/issues/3104) right, it seems it's still the case that it causes significant perf impact. Happy to learn otherwise.